### DISCUSSION TITLE
Extending with Javascript: File Management

### DISCUSSION THREAD

**Discussion by Mashiane** [pocketbase/pocketbase#3956] [Q&A]
Hi

I am interested in functionality of handling files withing the PocketBase WebServer structure e.g. files in assets folder of my deployed application. I have published an app in the "public" folder in PocketBase and am wondering if I can manipulate files in the /assets folder of my webapp.

For example, I want to create a folder, upload files to it, create a text file, append to it or delete it.

For example in php one is able to execute all kinds of file management functions and am wondering if its possible to do same here perhaps using "extensions" of some sort.

If you have any advice on the issue it could be appreciated.

Thank you so much.

PS: The files in question are not basically linked to collections but just stand alone files etc.

**Comment by ganigeorgiev** [comment]
If you want to interact directly with the local filesystem you can try the exposed [`$os.*` utils](https://pocketbase.io/jsvm/modules/_os.html) but I'm not sure if this will work for your use case as they are directly translated from the Go code and may not be very user friendly within the JS hooks context (_especially reading and writing files since you may have to call `close()` or similar on the open descriptors_).
Also note that if you use relative paths, keep in mind that they are relative to the running executable and not to the hooks files.
